canvas_sdk/effects/note/message.py

```diff
@@ -0,0 +1,107 @@
+import json
+from typing import Any
+from uuid import UUID
+
+from pydantic_core import InitErrorDetails
+
+from canvas_generated.messages.effects_pb2 import Effect
+from canvas_sdk.effects.note.base import NoteOrAppointmentABC
+from canvas_sdk.v1.data import Message as MessageModel
+from canvas_sdk.v1.data import Patient, Staff
+
+
+class Message(NoteOrAppointmentABC):
+    """
+    Effect to create and/or send a message.
+    """
+
+    class Meta:
+        effect_type = "MESSAGE"
+
+    message_id: str | UUID | None
+    content: str
+    sender_id: str | UUID
+    recipient_id: str | UUID
+    note_id: str | UUID | None
+    read: bool = False
+
+    def _get_error_details(self, method: Any) -> list[InitErrorDetails]:
+        errors = super()._get_error_details(method)
+
+        if (
+            not Patient.objects.filter(id=self.sender_id).exists()
+            or not Staff.objects.filter(id=self.sender_id).exists()
```

Comment:
My brain may not be working as I read this but shouldnt this be AND logic? If the patient and staff doesnt exist then this sender doesnt exist? (and for receipient) 

also double confirming the `id` is the externally_exposable UUID right so we can guarantee it is unique across patient and staff?

Reply:
or also a customer was asking the other day if you can automate creating a draft message that hasnt been sent but then an action on the UI can be taken when the user is ready to send the message.

Reply:
> My brain may not be working as I read this but shouldnt this be AND logic? If the patient and staff doesnt exist then this sender doesnt exist? (and for receipient)

You're 100% correct, guess it was my brain that wasn't working 😓 



> also double confirming the id is the externally_exposable UUID right so we can guarantee it is unique across patient and staff?

Yes ma'am. 


> or also a customer was asking the other day if you can automate creating a draft message that hasnt been sent but then an action on the UI can be taken when the user is ready to send the message.

That would be just creating a message right?

Reply:
Let me re-reply to your review again now that I've reacquainted myself with the code.

> also double confirming the id is the externally_exposable UUID right so we can guarantee it is unique across patient and staff?

Just to be clear, the Patient and Staff don't have EEID's, just `key`. This is that `key`.


> or also a customer was asking the other day if you can automate creating a draft message that hasnt been sent but then an action on the UI can be taken when the user is ready to send the message.

You can if it's staff to patient. Not patient to staff

Reply:
Okay woo all good. Thanks for following up